aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1438,14 +1440,114 @@ impl AptosVM {
 
     fn process_validator_transaction(
         &self,
-        _resolver: &impl AptosMoveResolver,
-        _txn: ValidatorTransaction,
-        _log_context: &AdapterLogSchema,
-    ) -> (VMStatus, VMOutput) {
-        (
-            VMStatus::Executed,
-            VMOutput::empty_with_status(TransactionStatus::Keep(ExecutionStatus::Success)),
-        )
+        resolver: &impl AptosMoveResolver,
+        txn: ValidatorTransaction,
+        log_context: &AdapterLogSchema,
+    ) -> Result<(VMStatus, VMOutput), VMStatus> {
+        let session_id = SessionId::validator_txn(&txn);
+        match txn {
+            ValidatorTransaction::DKGResult(dkg_node) => {
+                self.process_dkg_result(resolver, log_context, session_id, dkg_node)
+            },
+            ValidatorTransaction::DummyTopic1(_) => Ok((
+                VMStatus::Executed,
+                VMOutput::empty_with_status(TransactionStatus::Keep(ExecutionStatus::Success)),
+            )),
+            ValidatorTransaction::DummyTopic2(_) => Ok((
+                VMStatus::Executed,
+                VMOutput::empty_with_status(TransactionStatus::Keep(ExecutionStatus::Success)),
+            )),
+        }
+    }
+
+    fn process_dkg_result(
+        &self,
+        resolver: &impl AptosMoveResolver,
+        log_context: &AdapterLogSchema,
+        session_id: SessionId,
+        dkg_node: DKGNode,
+    ) -> Result<(VMStatus, VMOutput), VMStatus> {
+        //TODO(zjma): replace `StatusCode::INVALID_SIGNATURE` in this function
+        let dkg_state = load_on_chain_config_from_resolver::<DKGState>(resolver)
+            .map_err(|e| {
+                VMStatus::error(
+                    StatusCode::INVALID_SIGNATURE,
```

Comment:
@vgao1996 need some suggestions here.

Reply:
spoke with @georgemitenkov , it seems `INVALID_SIGNATURE` is not too bad...